src/mlpack/methods/ann/rbm/spike_slab_rbm_policy_impl.hpp

```diff
@@ -0,0 +1,343 @@
+
+#ifndef MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+#define MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+
+#include "spike_slab_rbm_policy.hpp"
+
+namespace mlpack {
+namespace ann {
+template<typename DataType>
+inline SpikeSlabRBMPolicy<DataType>
+      ::SpikeSlabRBMPolicy(const size_t visibleSize,
+      const size_t hiddenSize,
+      const size_t poolSize,
+      const ElemType slabPenalty, 
+      ElemType radius):
+      visibleSize(visibleSize),
+      hiddenSize(hiddenSize),
+      poolSize(poolSize),
+      slabPenalty(slabPenalty),
+      radius(radius)
+{
+  parameter.set_size(visibleSize * hiddenSize * poolSize +
+      visibleSize + hiddenSize);
+
+  visibleMean.set_size(visibleSize, 1);
+  spikeMean.set_size(hiddenSize, 1);
+  spikeSamples.set_size(hiddenSize, 1);
+  slabMean.set_size(poolSize, hiddenSize);
+  invSlabPenalty = 1.0 / slabPenalty;
+};
+
+// Reset function
+template<typename DataType>
+inline void SpikeSlabRBMPolicy<DataType>::Reset()
+{
+  // Weight shape D * K * N
+  weight = arma::cube(parameter.memptr(), visibleSize, poolSize, hiddenSize,
+      false, false);
+
+  // spike bias shape N * 1
+  spikeBias = DataType(parameter.memptr() + weight.n_elem, hiddenSize, 1,
+      false, false);
+
+  // visible penalty 1 * 1 => D * D(when used)
+  visiblePenalty = DataType(parameter.memptr() + weight.n_elem +
+      spikeBias.n_elem, 1, 1, false, false);
+}
+
+/**
+ * Free energy of the spike and slab variable
+ * the free energy of the ssRBM is given my
+ * $v^t$$\Delta$v - $\sum_{i=1}^N$ 
+ * $\log{ \sqrt{\frac{(-2\pi)^K}{\prod_{m=1}^{K}(\alpha_i)_m}}}$ -
+ * $\sum_{i=1}^N \log(1+\exp( b_i +
+ * \sum_{m=1}^k \frac{(v(w_i)_m^t)^2}{2(\alpha_i)_m})$
+ *
+ * @param input the visible layer
+ */ 
+template<typename DataType>
+inline typename SpikeSlabRBMPolicy<DataType>::ElemType
+SpikeSlabRBMPolicy<DataType>::FreeEnergy(DataType&& input)
+{
+  assert(input.n_rows == visibleSize);
+  assert(input.n_cols == 1);
+
+
+  ElemType freeEnergy = 0.5 * arma::as_scalar(visiblePenalty(0) * input.t() *
+      input);
+
+  freeEnergy -= 0.5 * hiddenSize * poolSize *
+      std::log((2.0 * M_PI) / slabPenalty);
+
+  for (size_t i = 0; i < hiddenSize; i++)
+  {
+    ElemType sum = 0;
+
+    for (size_t k = 0; k < poolSize; k++)
+    {
+      sum += arma::as_scalar(input.t() * weight.slice(i).col(k)) *
+          arma::as_scalar(input.t() * weight.slice(i).col(k)) /
+          (2.0 * slabPenalty);
+    }
+
+    freeEnergy -= SoftplusFunction::Fn(spikeBias(i) - sum);
+  }
+
+  return freeEnergy;
+}
+
+template<typename DataType>
+inline typename SpikeSlabRBMPolicy<DataType>::ElemType
+SpikeSlabRBMPolicy<DataType>::SpikeSlabRBMPolicy<DataType>
```

Comment:
That's why the Travis build errored.
`SpikeSlabRBMPolicy<DataType>::SpikeSlabRBMPolicy<DataType>::Evaluate`